src/store/modules/records/mutations.js

```diff
@@ -0,0 +1,87 @@
+import debug from 'debug'
+import Vue from 'vue'
+
+const logger = debug('app:store:records:mutations')
+const logMutation = (mutationName, ...args) => {
+  logger(`${mutationName} mutation being executed`, ...args)
+}
+
+export default {
+  SET_RECORDS(currentState, { listName, records }) {
+    logMutation('SET_RECORDS', listName, records)
+
+    // Filter out any records that don't have metadata attached to them.
+    //  These are records that were created by other providers.
+    currentState[listName] = records.filter((record) => {
+      return !!record.metadata
+    }) || []
+  },
+
+  ADD_RECORD_TO_LIST(currentState, { listName, record }) {
+    logMutation('ADD_RECORD_TO_LIST', listName, record)
+
+    // Filter out any records that don't have metadata attached to them.
+    //  These are records that were created by other providers.
+    if (!record.metadata) {
+      return
+    }
+
+    const exists = currentState[listName].find((existingRecord) => {
+      return existingRecord.tokenId === record.tokenId
+    })
+
+    if (!exists) {
+      currentState[listName].push(record)
+    }
+  },
+
+  REMOVE_RECORD_FROM_LIST(currentState, { listName, record }) {
+    logMutation('REMOVE_RECORD_FROM_LIST', listName, record)
+
+    currentState[listName] = currentState[listName].filter((codexRecord) => {
+      return codexRecord.tokenId !== record.tokenId
+    })
+  },
+
+  SET_ACTIVE_RECORD(currentState, record) {
+    logMutation('SET_ACTIVE_RECORD_INDEX', record)
+
+    currentState.activeRecord = record
+  },
+
+  UPDATE_RECORD_IN_LISTS(currentState, record) {
+    logMutation('UPDATE_RECORD_IN_LISTS', record)
+
+    // Filter out any records that don't have metadata attached to them.
+    //  These are records that were created by other providers.
+    if (!record.metadata) {
+      return
+    }
+
+    const listNames = [
+      'userRecords',
+      'incomingTransfers',
+      'outgoingTransfers',
+    ]
+
+    // Update the record in each list that we've cached
+    listNames.forEach((listName) => {
+      let index = -1
+      for (let i = 0; i < currentState[listName].length; i++) {
```

Comment:
This is basically a polyfill for `Array.prototype.findIndex`, why not just use that?

```javascript
listNames.forEach((listName) => {

  const listRecords = currentState[listName]

  const index = listRecords.findIndex((listRecord) => {
    return listRecord.tokenId === record.tokenId
  })

  if (index !== -1) {
    Vue.set(list, index, record)
  }
})
```

Reply:
Hah, I was looking up and down for that function so I just wrote it. I like that one better, thanks.